### PR TITLE
Tiebraker and Emergency execution committees

### PR DESCRIPTION
**Pull request by bulbozaur**

### CODE REVIEW

File: contracts/EmergencyActivationMultisig.sol [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    constructor(
+        address _owner,
+        address[] memory _members,
+        uint256 _quorum,
+        address _emergencyProtectedTimelock
+    ) RestrictedMultisigBase(_owner, _members, _quorum) {
+        emergencyProtectedTimelock = _emergencyProtectedTimelock;
+    }
+
+    function voteEmergencyActivate() public onlyMember {
```
Comment (skozin) [inline contracts/EmergencyActivationMultisig.sol:24]: i'd either not call this `vote` or rename the `vote` in the `EmergencyExecutionMultisig` since they have a diff semantics (this one approved, the other one allows to approve or disapprove)

e.g. name this `approveEmergencyActivate` and leave the other one `voteExecuteProposal`

File: contracts/EmergencyExecutionMultisig.sol [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        return _getState(_buildExecuteProposalAction(_proposalId));
+    }
+
+    function executeProposal(uint256 _proposalId) public {
+        _execute(_buildExecuteProposalAction(_proposalId));
+    }
+
+    // Governance reset
+
+    function voteGoveranaceReset() public onlyMember {
```
Comment (skozin) [inline contracts/EmergencyExecutionMultisig.sol:44]: i'd rename this to `approveGovernanceReset` (see the comment for `approveEmergencyActivate`)

File: contracts/RestrictedMultisigBase.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        }
+    }
+
+    function _vote(Action memory _action, bool _supports) internal {
+        bytes32 actionHash = _hashAction(_action);
+        if (actions[actionHash].data.length == 0) {
+            actions[actionHash].actionType = _action.actionType;
+            actions[actionHash].data = _action.data;
+        } else {
+            _checkStoredAction(_action);
```
Comment (skozin) [inline contracts/RestrictedMultisigBase.sol:41]: why is this needed given that we already hash the action when storing, and the action type+data is immutable?

File: contracts/TiebreakerSubDAO.sol [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    function _issueCalls(Action memory _action) internal override {
+        if (_action.actionType == APPROVE_PROPOSAL) {
+            uint256 proposalIdToExecute = abi.decode(_action.data, (uint256));
+            ITiebreakerCore(tiebreakerCore).voteApproveProposal(proposalIdToExecute, true);
+        } else {
+            assert(false);
+        }
+    }
+
+    function _buildApproveProposalAction(uint256 proposalId) internal view returns (Action memory) {
```
Comment (skozin) [inline contracts/TiebreakerSubDAO.sol:48]: i'd make this fn virtual in `TiebreakerCore`, make this class a subclass of `TiebreakerCore`, and override this fn and `_issueCalls` here; this way, we could avoid redefining `getApproveProposalState` and `approveProposal`

File: contracts/DualGovernance.sol [hunk 5]
```diff
@@ -143,19 +151,51 @@ contract DualGovernance is IGovernance, ConfigurationProvider {
...
-            _tiebreaker = newTiebreaker;
-            emit TiebreakerSet(newTiebreaker);
+        if (_tiebreaker == newTiebreaker) {
+            revert TieBreakerAddressIsSame();
+        }
+        if (_tiebreaker != address(0)) {
+            _proposers.unregister(CONFIG, _tiebreaker);
         }
+        _tiebreaker = newTiebreaker;
+        _proposers.register(newTiebreaker, CONFIG.ADMIN_EXECUTOR()); // TODO: check what executor should be. Reseal executor?
```
[the remark this answers is not in the document]
Comment (rkolpakov) [reply contracts/DualGovernance.sol:164]: Because we need to propose "resume" proposal and execute it. I agree that it's not the correct behavior so we need to specify the proposer for this case.

File: contracts/ExecutiveCommittee.sol [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    address public immutable OWNER;
+
+    address[] public membersList;
+    mapping(address => bool) public members;
+    uint256 public quorum;
+
+    mapping(bytes32 actionHash => ActionState) actionsStates;
+    mapping(address signer => mapping(bytes32 actionHash => bool support)) public approves;
```
[the remark this answers is not in the document]
Comment (bulbozaur) [reply contracts/ExecutiveCommittee.sol:45]: I can't see how should "voted against" be took into account while quorum calculating. 
+1 for "for", 0 for "no vote", -1 for "vote against"
looks unbalanced

File: contracts/ExecutiveCommittee.sol [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+    constructor(address owner, address[] memory newMembers, uint256 executionQuorum) {
+        if (executionQuorum == 0) {
+            revert InvalidQuorum();
+        }
+        quorum = executionQuorum;
```
[the remark this answers is not in the document]
Comment (bulbozaur) [reply contracts/ExecutiveCommittee.sol:50]: while adding/removing committee participants
